client/my-sites/domains/domain-management/components/primary-domain-selector/index.tsx

```diff
@@ -0,0 +1,217 @@
+
+	const shouldUpgradeToMakeDomainPrimary = ( domain: DomainData ): boolean => {
+		return (
+			! canUserSetPrimaryDomainOnThisSite &&
+			( domain.type === 'registered' || domain.type === 'mapping' ) &&
+			! domain.current_user_can_create_site_from_domain_only &&
+			! domain.wpcom_domain &&
+			! domain.is_wpcom_staging_domain &&
+			( site?.plan?.features?.active.includes( FEATURE_SET_PRIMARY_CUSTOM_DOMAIN ) ?? false )
+		);
+	};
+
+	let validPrimaryDomains = domains.filter( ( domain ) => {
+		return (
+			domain.can_set_as_primary &&
+			! domain.aftermarket_auction &&
+			domain.points_to_wpcom &&
+			! shouldUpgradeToMakeDomainPrimary( domain ) &&
+			! domain.primary_domain
+		);
+	} );
+
+	const hasWpcomStagingDomain = validPrimaryDomains.find(
+		( domain ) => domain.is_wpcom_staging_domain
+	);
+
+	if ( hasWpcomStagingDomain ) {
+		validPrimaryDomains = validPrimaryDomains.filter( ( domain ) => {
+			if ( domain.wpcom_domain ) {
+				return domain.is_wpcom_staging_domain;
+			}
+
+			return true;
+		} );
+	}
+
+	const onSelectChange = ( event: ChangeEvent< HTMLSelectElement > ) => {
+		setSelectedDomain( event.target.value );
+	};
+
+	const onSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		const domain = domains.find( ( d ) => d.domain === selectedDomain );
+		if ( ! domain || ! selectedDomain ) {
+			return;
+		}
+
+		setIsSettingPrimaryDomain( true );
+		onSetPrimaryDomain( selectedDomain, () => setIsSettingPrimaryDomain( false ), domain.type );
+	};
+
+	const trackUpgradeClick = ( isPlanUpgrade: boolean = false ) => {
+		recordTracksEvent( 'calypso_primary_site_address_nudge_cta_click', {
+			cta_name: isPlanUpgrade ? 'buy_a_plan' : 'add_custom_domain',
+		} );
+	};
+
+	if ( validPrimaryDomains.length > 0 && ! selectedDomain ) {
+		setSelectedDomain( validPrimaryDomains[ 0 ].domain );
+	}
+
+	return (
+		<Card className="domains-set-primary-address">
+			<CardHeading
+				className="domains-set-primary-address__title"
+				isBold={ true }
+				size={ 16 }
+				tagName="h2"
+			>
+				{ translate( 'Primary site address' ) }
+			</CardHeading>
+			<>
+				<p className="domains-set-primary-address__subtitle">
+					{ translate(
+						'The current primary address set for this site is: {{strong}}%(selectedDomain)s{{/strong}}.',
+						{
+							args: { selectedDomain: primaryDomain as string },
+							components: {
+								strong: <strong />,
+							},
+						}
+					) }{ ' ' }
+					{ ! canUserSetPrimaryDomainOnThisSite &&
+						translate(
+							"Your site plan doesn't allow to set a custom domain as a primary site address. {{planUpgradeLink}}Upgrade your plan{{/planUpgradeLink}}.",
+							{
+								components: {
+									planUpgradeLink: (
+										<a
+											href={ '/plans/' + primaryDomain }
+											onClick={ () => trackUpgradeClick( true ) }
+										/>
+									),
+								},
+							}
+						) }
+					{ validPrimaryDomains.length > 1 &&
```

Comment:
```suggestion
					{ validPrimaryDomains.length >= 1 &&
```
I think this needs to be `>= 1` because if the user has two domains and one of them is primary, the dropdown wouldn't be shown. Like in this screenshot:

<img width="1259" alt="Screenshot 2024-03-18 at 19 49 08" src="https://github.com/Automattic/wp-calypso/assets/5324818/8ef5ce78-7cb2-4f56-83ad-e96da2df8080">

Reply:
My bad, I was testing a PWPO with many domains and didn't noticed 🤦🏻‍♂️